BlockSettleUILib/OrderListModel.cpp

```diff
@@ -672,14 +672,21 @@ void OrderListModel::reset()
 
 void OrderListModel::processUpdateOrders(const Blocksettle::Communication::ProxyTerminalPb::Response_UpdateOrdersAndObligations &message)
 {
-   // Save latest selected index first
-   resetLatestChangedStatus(message);
-   // OrderListModel supposed to work correctly when orders states updated one by one.
-   // We don't use this anymore (server sends all active orders every time) so just clear old caches.
-   // Remove this if old behavior is needed
-   reset();
-   // Use some fake orderId so old code works correctly
-   int orderId = 0;
+   std::set<std::string> newIds;
+   for (const auto &data : message.orders()) {
+      newIds.insert(data.id());
+   }
+   bool orderRemoved = false;
+   for (const auto &knownId : knownIds_) {
+      if (newIds.find(knownId) == newIds.end()) {
+         orderRemoved = true;
+         break;
+      }
+   }
+   knownIds_ = std::move(newIds);
+   if (orderRemoved) {
+      reset();
```

Comment:
Not sure I understand this properly, but why we reset the whole model again instead of removing one line?

Reply:
this will happen only when some order removed, and orders removed once in 24 hours so not a problem
and adding new orders will work as expected